gateway/webhook/integration_test.go

```diff
@@ -175,27 +181,39 @@ func TestIntegrationWebhook(t *testing.T) {
 	}, time.Millisecond*500, time.Millisecond)
 
 	for i, tc := range testSetup.Cases {
-		if tc.Skip != "" {
-			t.Skip(tc.Skip)
-			continue
-		}
 		sConfig := sourceConfigs[i]
 
 		writeKey := sConfig.WriteKey
 		sourceID := sConfig.ID
 		workspaceID := sConfig.WorkspaceID
 
 		t.Run(tc.Name+"/"+tc.Description, func(t *testing.T) {
+			if tc.Skip != "" {
+				t.Skip(tc.Skip)
+				return
+			}
 			t.Logf("writeKey: %s", writeKey)
 			t.Logf("sourceID: %s", sourceID)
 			t.Logf("workspaceID: %s", workspaceID)
 
 			query, err := url.ParseQuery(tc.Input.Request.RawQuery)
+			// parse query parameters from input request
+			qParams := gjson.GetBytes(tc.Input.Request.Body, "query_parameters").Map()
+			if len(qParams) != 0 {
+				for k, v := range qParams {
+					vStr := v.Array()[0].String()
+					query.Set(k, vStr)
+				}
+			}
 			require.NoError(t, err)
 			query.Set("writeKey", writeKey)
 
 			t.Log("Request URL:", fmt.Sprintf("%s/v1/webhook?%s", gwURL, query.Encode()))
-			req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/v1/webhook?%s", gwURL, query.Encode()), bytes.NewBuffer(tc.Input.Request.Body))
+			method := http.MethodPost
+			if tc.Input.Request.Method == "GET" {
+				method = http.MethodGet
+			}
```

Comment:
simplify the logic to support all methods, defaulting to POST if empty
```suggestion
            method = tc.Input.Request.Method
			if method == "" {
			    method := http.MethodPost
			}
```